compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -2920,6 +2920,12 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     rustc_transmute::Reason::DstIsTooBig => {
                         format!("The size of `{src}` is smaller than the size of `{dst}`")
                     }
+                    rustc_transmute::Reason::SrcSizeOverflow => {
+                        format!("the size of `{src}` is overflow")
```

Comment:
Can you make this more like `LayoutError::SizeOverflow`?

```suggestion
                        format!("values of the type `{src}` are too big for the current architecture")
```

Same for dst.